flex-squid/src/main/java/org/sonar/flex/metrics/FileMetrics.java

```diff
@@ -50,7 +53,19 @@ public FileMetrics(FlexVisitorContext context) {
       FlexGrammar.RETURN_STATEMENT,
       FlexGrammar.THROW_STATEMENT,
       FlexGrammar.TRY_STATEMENT,
-      FlexGrammar.EMPTY_STATEMENT).size();
+      FlexGrammar.EMPTY_STATEMENT);
+
+    Set<Integer> alreadyMarked = new HashSet<>();
+    StringBuilder sb = new StringBuilder();
+    for (AstNode descendant : descendants) {
+      int line = descendant.getTokenLine();
+      if (alreadyMarked.add(line)) {
```

Comment:
This is not tested.

Reply:
@pynicolas test added in https://github.com/SonarSource/sonar-flex/pull/73/commits/80e609d8038bb9e8ced8130e7cdb4b6d1b3adf24.